drivers/shared/executor/executor.go

```diff
@@ -567,6 +561,14 @@ func lookupBin(taskDir string, bin string) (string, error) {
 		return root, nil
 	}
 
+	// when checking host paths, check with Stat first if path is absolute
+	// as exec.LookPath only considers files already marked as executable
+	// and only consider this for absolute paths to avoid depending on
+	// current directory of nomad which may cause unexpected behavior
+	if _, err := os.Stat(bin); err == nil && filepath.IsAbs(bin) {
```

Comment:
I considered allowing relative paths containing `/` that don't require search through PATH; however, I felt that reliance on nomad process current directory can be risky and uninuitive.  Given that this is a fix for a bug in 0.8, I felt it's fine to start with absolute paths first, then support other types of path if it's an issue.